### DISCUSSION TITLE
API Rules for group users.

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#2848] [Q&A]
I have two collections with fields details provided below
Group Collection
- id
- owner(User Relation Single Record)
- members(User Relation Multi Record)
Post Collection
-id
-title
-description
-isRestricted(bool)
-restrictedGroups(Multi Group Relation field)
User can create a post and he/she can choose post can be restricted to particular groups can view the post. Based on the selected groups, members under that group alone should see the post.

How to write API rule for this scenario where only user can view all allowed posts(Both Unresticted as well as posts that are restricted to group they are in)?

**Comment by ganigeorgiev** [comment]
If I'm understanding correctly the described structure, I think the below should work in you case:

```js
@request.auth.id != "" &&
restrictedGroups.members.id ?= @request.auth.id
```

The above roughly translates to: _"Require the user to be authenticated and at least one of the post groups to have at least one member.id that is equal to the user.id"_

**Reply by iamdineshkumar** [reply]
Hi 
I have a new case related to this scope
Now there is additional collection as category with following fields
-Category
-owner(User Relation Single Record-> only owner can able to edit this record)


Post Collection is modified as below
-id
-category(Category collection relation single record)
-title
-description
-isRestricted(bool)
-restrictedGroups(Multi Group Relation field)

Since each users can have their own categories and can update them as well as other person categories should not be shown to them. So for categories API rule for view & listing will be
``` js
@request.auth.id ! ="" &&  @request.auth.id  =owner 
```

But for post restrictedGroup user should know what category under which post is made by expanding category field. Since they are not having rights as per above rule, not getting response. How to give API rules of category so that in post can be expanded to get category details in case member belong to restrictedGroups.members of that post?

**Reply by ganigeorgiev** [reply]
I'm not entirely sure that I understand the requirements and the expected outcome, but you can't disable the view category access and at the same time allow to be expanded.

If you want the categories to be expandable by the restricted group members you'll need to perform a similar check for the posts from the category rule. One way to do that could be to join the posts collection with `@collection.posts.*` and use a category view rule like:

```js
@request.auth.id != "" &&
(
  @request.auth.id = owner ||
  (@collection.posts.category ?= id && @collection.posts.restrictedGroups.members.id ?= @request.auth.id)
)
```

_(keep in mind that joins with large tables may not be very efficient)_

**Reply by iamdineshkumar** [reply]
I just thought about another way, instead of restring both listing & view API rule, if I restrict only listing with owner check means whether category can be expanded during query post?

**Reply by ganigeorgiev** [reply]
Yes, when expanding relation we check only the related collection View API rule.
